TsRandomizer/Randomisation/Gate.cs

```diff
@@ -1,14 +1,40 @@
-﻿using System.Linq;
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
 using TsRandomizer.Extensions;
 
 namespace TsRandomizer.Randomisation
 {
-	abstract class Gate
+	abstract class Gate : IEquatable<Gate>
 	{
 		public abstract bool CanBeOpenedWith(Requirement obtainedRequirements);
 
 		public abstract bool Requires(Requirement requirementsToCheck);
 
+		private static List<Gate> requirementGates = new List<Gate>();
```

Comment:
Plz try to avoid using static variables that you keep changing, as they never reset state. they will maintain thier value even across rando games aslong as you keep the TsRandomizer.exe open